metrics/sink.go

```diff
@@ -15,143 +15,186 @@ var (
 	_ Sink = &DummySink{}
 )
 
+// Sink collects Sample's values.
 type Sink interface {
-	Add(s Sample)                              // Add a sample to the sink.
-	Format(t time.Duration) map[string]float64 // Data for thresholds.
-	IsEmpty() bool                             // Check if the Sink is empty.
+	// Add adds a sample to the sink.
+	Add(s Sample)
+
+	// IsEmpty checks if the Sink has never received a value.
+	IsEmpty() bool
 }
 
 type CounterSink struct {
-	Value float64
-	First time.Time
+	value float64
+	first time.Time
 }
 
 func (c *CounterSink) Add(s Sample) {
-	c.Value += s.Value
-	if c.First.IsZero() {
-		c.First = s.Time
+	c.value += s.Value
+	if c.first.IsZero() {
+		c.first = s.Time
 	}
 }
 
+// LastValue returns the counter value.
+func (c *CounterSink) LastValue() float64 {
+	return c.value
+}
+
 // IsEmpty indicates whether the CounterSink is empty.
-func (c *CounterSink) IsEmpty() bool { return c.First.IsZero() }
+func (c *CounterSink) IsEmpty() bool { return c.first.IsZero() }
 
-func (c *CounterSink) Format(t time.Duration) map[string]float64 {
-	return map[string]float64{
-		"count": c.Value,
-		"rate":  c.Value / (float64(t) / float64(time.Second)),
+// Rate computes the rate per second of the aggregated values.
+func (c *CounterSink) Rate(t time.Duration) float64 {
```

Comment:
It seems like it should be a helper function, not a method of the `Counter` type :thinking: 

Most of the complexity here is in dealing with the external parameter `t`, which has nothing to do with the `value`. Everything it needs from the `CounterSink` it can get from its public methods.